components/Multiselector.jsx

```diff
@@ -0,0 +1,111 @@
+import { useRef } from "react";
+import Multiselect from "multiselect-react-dropdown";
+import styled from "styled-components";
+import { baseTheme } from "../theme";
+
+/*
+    params: 
+    - options = available items to choose from
+    - displayValue = property name to display in the dropdown options
+    - placeholder = placeholder in the input section
+    - setList = function that will update the listType variable (holds all valid tags)
+    - searchValue = current string of search field input
+    - tagValue = current array of selected tags
+    - otherValue = current array of selected values (blog --> releaseBatch, opp --> city)
+    - handleChange = function in main jsx file that will execute every change 
+    - isFilterTag = boolean indicating if the filter is for tags or otherValue
+*/
+export const Multiselector = ({
+  options,
+  displayValue,
+  placeholder,
+  setList,
+  searchValue,
+  tagValue,
+  otherValue,
+  handleChange,
+  isFilterTag = true,
+  isObject = false,
+  emptyRecordMsg = "No results found",
+  avoidHighlightFirstOption = true,
+  showArrow = true,
+  ...props
+}) => {
+  const multiselectRef = useRef([]);
+  const handleAdd = () => {
+    setList(multiselectRef.current.getSelectedItems());
+    if (isFilterTag) {
```

Comment:
That solution will also prevent this type of code from getting in and so we don't have to use if statements